modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/exportimport/data/handler/DEDataDefinitionFieldLinkStagedModelDataHandler.java

```diff
@@ -146,19 +147,44 @@ protected void doImportStagedModel(
 					"link-class-name")));
 		importedDEDataDefinitionFieldLink.setDdmStructureId(ddmStructureId);
 
-		DDMStructure ddmStructure = _ddmStructureLocalService.getDDMStructure(
-			ddmStructureId);
+		if (StringUtil.equals(
+				importedDEDataDefinitionFieldLink.getClassName(),
+				DDMStructureLayout.class.getName())) {
 
-		DDMStructureVersion structureVersion =
-			ddmStructure.getStructureVersion();
+			Map<Long, Long> structureLayoutIds =
+				(Map<Long, Long>)portletDataContext.getNewPrimaryKeysMap(
+					DDMStructureLayout.class);
 
-		DDMStructureLayout structureLayout =
-			_ddmStructureLayoutLocalService.
-				getStructureLayoutByStructureVersionId(
-					structureVersion.getStructureVersionId());
+			DDMStructureLayout structureLayout =
+				_ddmStructureLayoutLocalService.getStructureLayout(
```

Comment:
Hey, @IstvanD. Here you're relying on having the original structure layout stored in the same database. This won't work if you're importing the LAR to a completely different DXP instance.
You need to come up with a solution that doesn't involve searching for the original data.
Feel free to reach out over Slack if you find difficulties.

We really appreciate the effort you're putting into this.